### PR TITLE
Do not print warning message for +CmiSleepOnIdle when arg is passed

### PR DESCRIPTION
**Pull request by nitbhat**
Additionally, this commit also modifies the warning message printed
since the previous message was confusing.

### CODE REVIEW

File: src/ck-core/init.C [hunk 1]
```diff
@@ -1638,10 +1638,10 @@ void _initCharm(int unused_argc, char **argv)
             // XXX: Assuming uniformity of node size here
             num += num/CmiMyNodeSize();
 #endif
-            if (!_Cmi_forceSpinOnIdle && num > CmiNumCores())
+            if (!_Cmi_sleepOnIdle && !_Cmi_forceSpinOnIdle && num > CmiNumCores())
             {
               if (CmiMyPe() == 0)
-                CmiPrintf("\nCharm++> Warning: the number of SMP threads (%d) is greater than the number of physical cores (%d), so threads will sleep while idling. Use +CmiSpinOnIdle or +CmiSleepOnIdle to control this directly.\n\n", num, CmiNumCores());
+                CmiPrintf("\nCharm++> Warning: the number of SMP threads (%d) is greater than the number of physical cores (%d), so internally setting +CmiSleepOnIdle to make threads sleep while idling. Use +CmiSpinOnIdle to change this behavior and make the runtime system spin on message reception when idle, rather than sleeping\n\n", num, CmiNumCores());
```
Comment (matthiasdiener) [inline src/ck-core/init.C]: Could add a '.' at the end of the string.

**Comment by evan-charmworks** [comment]
I definitely support adding `!_Cmi_sleepOnIdle` to the if statement. I feel like the new message text is a bit wordy though, and could be made more concise. I don't think we need to fully explain these options in the warning text.

My suggestion:

```
"Charm++> Warning: Running with more SMP threads (%d) than physical cores (%d).\n"
"         Pass +CmiSleepOnIdle (default behavior) or +CmiSpinOnIdle to silence this message.\n"
```

This would match the message when running a Charm program without any provisioning arguments:
```
Charm++> No provisioning arguments specified. Running with a single PE.
         Use +auto-provision to fully subscribe resources or +p1 to silence this message.
```

**Comment by nitbhat** [comment]
> Neither of these options is described in the manual, but this could be added in another PR.

I've added this issue (https://github.com/UIUC-PPL/charm/issues/2747) to have all/most runtime flags documented.
